feat: expose webpack target via babel caller (#826)
babel/babel-loader: README.md

  }
```

## Customize config for specific Webpack target

> Webpack supports bundling for multiple different [targets](https://webpack.js.org/concepts/targets/) including the web, and node. Sometimes it can be useful to have a slightly different babel configuration for the different environments. This loader provides a `target` property via the babel [caller](https://babeljs.io/docs/en/config-files#apicallercb) API which can be used in this case:

Reviewer: ```suggestion
Webpack supports bundling for multiple [targets](https://webpack.js.org/concepts/targets/). For cases where you may want different Babel configurations for each target (like `web` _and_ `node`), this loader provides a `target` property via Babel's [caller](https://babeljs.io/docs/en/config-files#apicallercb) API:
```

```javascript
// babel.config.js

module.exports = api => {
  return {
    plugins: [
      "@babel/plugin-proposal-nullish-coalescing-operator",
      "@babel/plugin-proposal-optional-chaining"
    ],
    presets: [
      [
        "@babel/preset-env",
        {
          useBuiltIns: "entry",
          // caller.target will be the same as the target option from webpack
          ...(api.caller(caller => caller.target === "node")

>             ? { node: "current" }

Reviewer: Shouldn't this also be `{ targets: { node: 'current' }}`?

Author: Yeup, you're right. I've gone ahead and fixed that up. Thanks!

            : { targets: { chrome: "58", ie: "11" } }
          )
        }
      ]
    ]
  }
}
```

> In the above `babel.config.js` we're able to distinguish if we are transpiling for `node` or the `web` and adjust accordingly. In this case it has been configured such that `node` targets use `@babel/preset-env` to transpile features for the current version of node, while the `web` target will transpile features back to ie 11 and Chrome 58.

Reviewer: I feel like we can skip explaining the targets with a quick description of the example (maybe with something like what I suggested above)?

Thoughts?

Author: I agree. I have incorporated these changes in the PR. Thanks for the feedback 😄
